test-e2e.sh

```diff
@@ -1,33 +1,57 @@
-#!/bin/bash
+#!/usr/bin/env bash
 
-set -e
+set -eu -o pipefail
 
-go build -o ./dist/klone_test .
-klone_binary=$(realpath ./dist/klone_test)
+klone_binary=$(realpath $1)
+
+# 1. Test that the sync command works for the example directory
 
-# Test that the sync command works for the example directory
 pushd ./example
 "$klone_binary" sync
 popd
 
-# Create a temp directory and test that the init, add, and sync commands work
+# 2. Create a temp directory and test that the init, add, and sync commands work
+
 temp_dir=$(mktemp -d)
 trap '{ rm -rf "$temp_dir"; echo "> Deleted temp dir $temp_dir"; }' EXIT
+
 pushd "$temp_dir"
+
 "$klone_binary" init
+
 mkdir -p a/b/e
+
 touch a/SHOULD_NOT_BE_DELETED
 touch a/b/SHOULD_BE_DELETED
 touch a/b/e/SHOULD_BE_DELETED
+
 "$klone_binary" add a/b c/d https://github.com/cert-manager/community.git logo main
+
 "$klone_binary" add a/b e https://github.com/cert-manager/community.git logo main 9f0ea0341816665feadcdcfb7744f4245604ab28
+
 "$klone_binary" sync
-if [ -f a/SHOULD_NOT_BE_DELETED ] && [ ! -f a/b/SHOULD_BE_DELETED ]; then
-  echo "Test passed"
-else
-  echo "Test failed"
-  exit 1
+
+if [ ! -f a/SHOULD_NOT_BE_DELETED ]; then
+	echo "Test failed: a/SHOULD_NOT_BE_DELETED not found"
+	exit 1
+fi
+
+if  [ -f a/b/SHOULD_BE_DELETED ]; then
+	echo "Test failed: a/b/SHOULD_BE_DELETED was found"
+	exit 1
+fi
+
+if  [ -f a/b/e/SHOULD_BE_DELETED ]; then
+	echo "Test failed: a/b/e/SHOULD_BE_DELETED was found"
+	exit 1
 fi
```

Comment:
note: this check was missing previously, I think it was missed in #11

Reply:
It was something I used to check manually when running the test, thanks for automating this.

Reply:
Indeed, thanks for noting this!